sway/input/seat.c

```diff
@@ -71,6 +72,21 @@ static void seat_node_destroy(struct sway_seat_node *seat_node) {
 	free(seat_node);
 }
 
+void seat_idle_notify_activity(struct sway_seat *seat,
+		enum sway_input_idle_source source) {
+	uint32_t mask = seat->idle_inhibit_sources;
+	struct wlr_idle_timeout *timeout;
+	wl_list_for_each(timeout, &server.idle->idle_timers, link) {
+		if (timeout->idle_state) {
+			mask = seat->idle_wake_sources;
```

Comment:
This doesn't work well with multiple timers, each having a different timeout. A 5s screen dim timeout combined with a 30s dpms timeout would break this for instance.

Reply:
The idea behind this code was just to find out if any timer was idle, which it does. So, by design?

Reply:
I'm just saying this design breaks timers that are not idle (going to sleep while they shouldn't).

Reply:
I don't understand.

Reply:
Let's say touch is not in `idle_wake_sources`. Let's say you want to setup one timer to dim the screen after 5s and another timer to blank the screen after 30s. It's a pretty standard setup people have on phones.

You stop touching the screen for 10s. The screen is dimmed. Then you start touching the screen again. Because the dim timer has expired, the touches don't wake timers up to un-dim the screen. Worse, the blank timer isn't ping'ed and no matter how much you touch your screen it'll blank the screen after 20 more seconds.

Reply:
Got it, thanks for explaining. I'll update this to kick in once all idle timers have expired instead.

Reply:
Yeah, that's probably a good solution. I was thinking about more complicated solutions as usual :P

Reply:
Maybe at some point we'll need to get to the more complicated solutions though. If you add in a messaging client using the idle protocol to mark the user as away after 1min, things get more complicated. Needs more thought.